src/Lecture4.hs

```diff
@@ -237,7 +299,40 @@ you can return string "no value"
 -}
 
 displayStats :: Stats -> String
-displayStats = error "TODO"
+displayStats stats =
+      "Total positions:       : "
+   <> showGetSum (statsTotalPositions stats)
+   <> "\n"
+   <> "Total final balance    : "
+   <> showGetSum (statsTotalSum stats)
+   <> "\n"
+   <> "Biggest absolute cost  : "
+   <> showGetMax (statsAbsoluteMax stats)
+   <> "\n"
+   <> "Smallest absolute cost : "
+   <> showGetMin (statsAbsoluteMin stats)
+   <> "\n"
+   <> "Max earning            : "
+   <> maybe "no value" showGetMax (statsSellMax stats)
+   <> "\n"
+   <> "Min earning            : "
+   <> maybe "no value" showGetMin (statsSellMin stats)
+   <> "\n"
+   <> "Max spending           : "
+   <> maybe "no value" showGetMax (statsBuyMax stats)
+   <> "\n"
+   <> "Min spending           : "
+   <> maybe "no value" showGetMin (statsBuyMin stats)
+   <> "\n"
+   <> "Longest product name   : "
+   <> unMaxLen (statsLongest stats)
```

Comment:
That's a neat implementation!
Alternatively, you can put each line as a separate element in the list and use `unlines`. So you don't need to write `"\n"` manually